CONTRIBUTING.md

```diff
@@ -0,0 +1,162 @@
+me angry thinking about it!
+
+There's a [detailed style guide](.copy-edit-stylesheet-checklist.md) if you
+want to know how to format things, and we run a spellchecker in our [CI
+system](.travis.yml) as well so you may need to add lines to
+[.aspell.en.pws].
+
+### Writing The Requirements
+
+Some requirements are obvious, some are subtle.  They're designed to walk
+an implementer through the code they have to write, so write them as YOU
+develop YOUR implementation.  Stick with `MUST`/`SHOULD`/`MAY` and `NOT`:
+see [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+Requirements are grouped into writer and reader, just as implementations
+are.  Make sure you define exactly what a writer must do, and exactly what
+a reader must do if it the writer doesn't do that!  A developer should
+never have to intuit reader requirements from writer ones.
+
+Note that the data doesn't have requirements: don't say `foo MUST be 0`,
+say `The writer MUST set foo to 0` and `The reader MUST fail the connection
+if foo is not 0`.
+
+Avoid the term `MUST check`: use `MUST fail the connection if` or `MUST
+fail the channel if` or `MUST send an error message if`.
+
+There's a subtle art here for future extensions: you might say `a writer
+MUST set foo to 0` and not mention it in the reader requirements, but it's
+better to say `a reader MUST ignore foo`.  A future version of the spec
+might define when a writer sets `foo` to `1` and we know that old readers
+will ignore it.
+
+`MAY` is a hint as to what something is for: an implementation may do
+anything not written in the spec anyway.  `MUST` is when not doing
+something will break the protocol or security.
+
+Requirements can be vague (eg. "in a timely manner"), but only as a last
+resort admission of defeat.  If you don't know, what hope has the poor
+implementer?
+
+### Creating Test Vectors
+
+For new low-level protocol constructions, test vectors are necessary.
+These have traditionally been lines within the spec itself, but the modern
+trend is to use JSON and separate files.  The intent is that they be
+machine-readable by implementations.
+
+For new inter-peer messages, a test framework is in development to simulate
+entire conversations.
+
+## Specification Modification Process
+
+There is a [mailing
+list](https://lists.linuxfoundation.org/mailman/listinfo/lightning-dev)
+for larger feature discussion, a [GitHub
+repository](https://github.com/lightningnetwork/lightning-rfc) for
+explicit issues and pull requests, and a bi-weekly IRC meeting on
+#lightning-dev on Freenode, currently held at 5:30am Tuesday, 
+Adelaide/Australia timezone (eg. Tuesday 23rd July 2019 05:30 == Mon, 22
+Jul 2019 20:00 UTC).
+
+Spelling, typo and formatting changes are accepted once two contributors
+ack and there are no nacks.  All other changes get approved and minuted at
+the IRC meeting.  Protocol changes require two independent implementations
+which successfully inter-operate.
```

Comment:
and maybe add a sentence inviting people to be patient and polite if they PR to correct spec oddities and remembering them there are multiples implementations to coordinate and that takes time.